addons/mrp/models/mrp_production.py

```diff
@@ -1226,13 +1226,15 @@ def action_view_mrp_production_backorders(self):
     def _prepare_stock_lot_values(self):
         self.ensure_one()
         name = self.env['ir.sequence'].next_by_code('stock.lot.serial')
-        exist_lot = self.env['stock.lot'].search([
+        next_serial = not name or self.env['stock.lot'].search([
```

Comment:
keep existing variables

Reply:
if name is False exist_lot is not really appropriate